### PR TITLE
feat: buffer-local var for lang override

### PR DESCRIPTION
**Pull request by rschmukler**
Adds functionality to override the linting language used by clj-kondo.

When connected to sibling cljs and clj repls in cider, it will use the
major mode to decide where to execute evaluated sexps. When working with
cljc files it can be useful to set the major mode of emacs to
clojure-mode or clojurescript-mode to restrict where the sexps are being
evaluated while you develop.

This PR adds the ability for a buffer to specify the language it would
like to be linted as, allowing cljc buffers being edited in clojure-mode
to still be linted by the cljc linter in clj-kondo.

### CODE REVIEW

**Comment by borkdude** [comment]
Reviewed by @sogaiu

**Comment by borkdude** [comment]
@rschmukler Looks good, although I wondered if `eval` was really necessary. I think `flycheck-clj-kondo-lang` is a better name than `flycheck-clj-kondo-lint-lang` since the option is called `lang` and not `lint-lang`.

**Comment by rschmukler** [comment]
@borkdude thanks for the review. I've updated the variable name. The `eval` is part of the `flycheck-substitute-argument` syntax. Just to be sure, I tried removing it and the macro fails to compile without it.

Thanks again!

File: flycheck-clj-kondo.el [hunk 1]
```diff
@@ -43,36 +43,44 @@
...
+      (warning line-start "<stdin>:" line ":" column ": " (0+ not-newline) "warning: " (message) line-end)
+      (info line-start "<stdin>:" line ":" column ": " (0+ not-newline) "info: " (message) line-end))
+     :modes (,mode)
+     :predicate (lambda ()
+                  (or
+                   ;; We are being told to explicitly lint
+                   flycheck-clj-kondo-lang
+                   ;; If there is an associated file with buffer, use file name extension
+                   ;; to infer which language to turn on.
+                   (and buffer-file-name
```
Comment (borkdude) [inline flycheck-clj-kondo.el:76]: This seems to be a change in logic. 

Before it was: if there is a buffer-file-name, then the predicate returns the value of `(string= ,lang (file-name-extension buffer-filename)`. 

So if the buffer-file-name condition doesn't hold, we fall back on the clojure mode detection. I think that's reasonable.

Just thinking out loud here :).

It seems this PR changes that from: both have to be true, else we fall back on comparing lang to the right clojure-mode.

**Pull request merged**